src/Sylius/Component/Product/Model/ProductVariant.php

```diff
@@ -36,6 +36,8 @@ class ProductVariant implements ProductVariantInterface
      */
     protected $code;
 
+    protected int $productId;
```

Comment:
I believe this additional field is not needed - we can join the product table in the query build if needed

Reply:
We really don't want any `JOIN` in this functionality. It's query overhead. Sometimes it's needed to have only id rather than a reference. I believe that accessing product id by relation will create inner/left join by Doctrine.

Reply:
However we don't do a database-driven development, we're designing the application and code to be nice and shiny and let ORM do its job 😄  This `productId` here is used **only** to simplify the query which cannot even overload the database too much as we're limiting the number of data taken from the database by default. If it would be a problem for anyone - one would just override the query and customize the entity in their application :dancer:

Reply:
Product variant is the owner of this relationship, so there's already a `product_id` field the database table `sylius_product_variant.`. You can do `$productVariant->getProduct()->getId()` to get the value and this won't perform any additional queries.

Reply:
Yes but Doctrine will generate JOIN as soon as it detects calling getter method in relation object.
`$productVariant->getProduct()->getId()` is ok to satisfy "we're designing the application and code to be nice and shiny".
Anyway, it's not a performance problem but rather a good practice. Calling getters in relation objects will always lead to very bad designed application in case of performance and there is almost no escape from it. I guess it's the problem of people using Sylius - not ours.

Reply:
You say it's not a performance problem, but then the very next sentence you say the opposite, which is confusing.
In this particular case, we're not calling any getters except `getId()`,a getter for the identifier, which is treated differently in Doctrine proxies. It will not generate any joins or other extra queries.

Reply:
Good then I didn't know it.